### ISSUE TITLE
Overcomplicated code in finding the GFWList files makes it fail to find the correct file in xray on NixOS

### ISSUE THREAD

**Issue by iopq** [v2rayA/v2rayA#1162]
### Version, approach of installation, operating system

1. What version of v2rayA are you using?
2.2.4

3. What approach of installation do you install V2rayA?
nixpkgs

5. What is your operating system?
NixOS

### Description
When I overlay xray into the package instead of v2ray, it fails to start with a message like 

```
[I] [io.go:429] Failed to start: main: failed to load config files: [/etc/v2raya/config.json] > infra/conf: invalid field rule > infra/conf: failed to load GeoIP: private > infra/conf: failed to load file: geoip.dat > infra/conf: failed to open file: geoip.dat > open /run/user/0/v2raya:/nix/store/hzadd4qz8q4h4c5bwmg6jadb3lid0mhm-v2ray-assets/share/v2ray/geoip.dat: no such file or directory
```

### Steps to reproduce
in NixOS, in overlays add `final: prev: { v2raya= prev.v2raya.override { v2ray = final.xray; }; })` and `sudo nixos-rebuild switch`

Go into v2raya and try to start xray, it will fail

### Logs：

 ```bash
 # daemon mode
 journalctl -u v2raya | tail -n 50
 sudo tail -n 50 /var/log/v2raya/v2raya.log
 
 # docker mode
 docker logs v2raya
 ```

 journalctl -u v2raya | tail -n 50
Nov 10 12:38:06 desktop systemd[1]: Started v2rayA service.
Nov 10 12:45:09 desktop v2rayA[108841]: Quitting...
Nov 10 12:45:09 desktop systemd[1]: Stopping v2rayA service...
Nov 10 12:45:09 desktop systemd[1]: v2raya.service: Deactivated successfully.
Nov 10 12:45:09 desktop systemd[1]: Stopped v2rayA service.
Nov 10 12:45:09 desktop systemd[1]: v2raya.service: Consumed 2.336s CPU time, received 75.6K IP traffic, sent 7.2M IP traffic.
Nov 10 12:47:24 desktop systemd[1]: Started v2rayA service.
Nov 10 12:50:42 desktop v2rayA[114530]: Quitting...
Nov 10 12:50:42 desktop systemd[1]: Stopping v2rayA service...
Nov 10 12:50:42 desktop systemd[1]: v2raya.service: Deactivated successfully.
Nov 10 12:50:42 desktop systemd[1]: Stopped v2rayA service.
Nov 10 12:50:42 desktop systemd[1]: v2raya.service: Consumed 5.062s CPU time, received 1.4M IP traffic, sent 3.3M IP traffic.
Nov 10 12:50:43 desktop systemd[1]: Started v2rayA service.
Nov 10 12:55:14 desktop v2rayA[115740]: Quitting...
Nov 10 12:55:14 desktop systemd[1]: Stopping v2rayA service...
Nov 10 12:55:14 desktop systemd[1]: v2raya.service: Deactivated successfully.
Nov 10 12:55:14 desktop systemd[1]: Stopped v2rayA service.
Nov 10 12:55:14 desktop systemd[1]: v2raya.service: Consumed 814ms CPU time, received 65.7K IP traffic, sent 5.4M IP traffic.
Nov 10 12:55:15 desktop systemd[1]: Started v2rayA service.
Nov 10 12:55:17 desktop v2rayA[116602]: Quitting...
Nov 10 12:55:17 desktop systemd[1]: Stopping v2rayA service...
Nov 10 12:55:18 desktop systemd[1]: v2raya.service: Deactivated successfully.
Nov 10 12:55:18 desktop systemd[1]: Stopped v2rayA service.
Nov 10 12:55:18 desktop systemd[1]: v2raya.service: Consumed 2.331s CPU time, received 10.2K IP traffic, sent 3.2K IP traffic.
Nov 10 12:55:19 desktop systemd[1]: Started v2rayA service.
Nov 10 12:55:21 desktop v2rayA[117374]: Quitting...
Nov 10 12:55:21 desktop systemd[1]: Stopping v2rayA service...
Nov 10 12:55:21 desktop systemd[1]: v2raya.service: Deactivated successfully.
Nov 10 12:55:21 desktop systemd[1]: Stopped v2rayA service.
Nov 10 12:55:21 desktop systemd[1]: v2raya.service: Consumed 685ms CPU time, received 7.6K IP traffic, sent 1.4K IP traffic.
Nov 10 13:07:50 desktop systemd[1]: Started v2rayA service.

#

2023/11/10 13:07:50.710 [A] [main.go:29] V2Ray binary is /nix/store/4laqk1bwna42rg4jx0qhr2hfkc8m924h-xray-1.8.4/bin/xray
2023/11/10 13:07:50.710 [A] [main.go:29] V2Ray asset directory is /run/user/0/v2raya
2023/11/10 13:07:50.710 [A] [main.go:29] v2rayA working directory is /
2023/11/10 13:07:50.710 [A] [main.go:29] v2rayA configuration directory is /etc/v2raya
2023/11/10 13:07:50.710 [A] [main.go:29] Golang: go1.21.3
2023/11/10 13:07:50.710 [A] [main.go:29] OS: linux
2023/11/10 13:07:50.710 [A] [main.go:29] Arch: amd64
2023/11/10 13:07:50.710 [A] [main.go:29] Lite: false
2023/11/10 13:07:50.710 [A] [main.go:29] Version: 2.2.4
2023/11/10 13:07:50.710 [A] [main.go:29] Starting...
2023/11/10 13:07:50.739 [I] [io.go:432] Xray 1.8.4 (Xray, Penetrates Everything.) Custom (go1.21.3 linux/amd64)
2023/11/10 13:07:50.739 [I] [io.go:432] A unified platform for anti-censorship.
2023/11/10 13:07:50.739 [I] [io.go:432] [Info] infra/conf/serial: Reading config: /etc/v2raya/config.json
2023/11/10 13:07:50.741 [I] [io.go:432] Failed to start: main: failed to load config files: [/etc/v2raya/config.json] > infra/conf: invalid field rule > infra/conf: failed to load GeoIP: private > infra/conf: failed to load file: geoip.dat > infra/conf: failed to open file: geoip.dat > open /run/user/0/v2raya:/nix/store/vlc67cd8q3w7aiwgfsyxywz4q6qhdjym-v2ray-assets/share/v2ray/geoip.dat: no such file or directory
2023/11/10 13:07:50.973 [W] [asm_amd64.s:1650] v2ray-core: exit status 23
2023/11/10 13:07:50.973 [E] [main.go:30] failed to start v2ray-core: unexpected exiting: check the log for more information
2023/11/10 13:07:50.981 [A] [index.go:116] v2rayA is listening at http://127.0.0.1:2017
2023/11/10 13:07:50.981 [A] [index.go:116] v2rayA is listening at http://[::1]:2017
2023/11/10 13:08:13.283 [I] [io.go:432] Xray 1.8.4 (Xray, Penetrates Everything.) Custom (go1.21.3 linux/amd64)
2023/11/10 13:08:13.283 [I] [io.go:432] A unified platform for anti-censorship.
2023/11/10 13:08:13.283 [I] [io.go:432] [Info] infra/conf/serial: Reading config: /etc/v2raya/config.json
2023/11/10 13:08:13.284 [I] [io.go:432] Failed to start: main: failed to load config files: [/etc/v2raya/config.json] > infra/conf: invalid field rule > infra/conf: failed to load GeoIP: private > infra/conf: failed to load file: geoip.dat > infra/conf: failed to open file: geoip.dat > open /run/user/0/v2raya:/nix/store/vlc67cd8q3w7aiwgfsyxywz4q6qhdjym-v2ray-assets/share/v2ray/geoip.dat: no such file or directory
2023/11/10 13:08:13.504 [W] [asm_amd64.s:1650] v2ray-core: exit status 23
2023/11/10 13:08:13.605 [E] [connection.go:88] failed to start v2ray-core: unexpected exiting: check the log for more information

```

**Comment by iopq** [comment]
I have fixed it in my fork with a hammer approach of removing all that code:

https://github.com/xbclub/xraya/compare/feat_v5...iopq:xraya:feat_v5#diff-d77caaf7273c910d994b3db314b9ddc744d47ce330d9e98e3b2cab04e9a9c45b

but it will probably break some stuff for other people on other distros

I just want it to read the CORRECT file that is already pulled in which is `/nix/store/vlc67cd8q3w7aiwgfsyxywz4q6qhdjym-v2ray-assets/share/v2ray/geoip.dat`

**Comment by iopq** [comment]
/run/user/0/v2raya folder also exists, not sure why the filename in the log says `/run/user/0/v2raya:/nix/store/vlc67cd8q3w7aiwgfsyxywz4q6qhdjym-v2ray-assets/share/v2ray/geoip.dat` which is a list of paths, not a path itself

**Comment by iopq** [comment]
Specifically the code in service/core/v2ray/asset/asset.go is a bit hard to follow

**Comment by guyangliu** [comment]
I've got same problem with you.
